eval/src/expr/if_expr.rs

```diff
@@ -0,0 +1,36 @@
+use parser::ast::Expression;
+
+use crate::{
+    context::Context,
+    errors::{EvalError, TypeError, TypeErrorKind},
+    jml_type::JmlType,
+    value::JmlValue,
+};
+
+use super::eval_expr;
+pub fn eval_if_expr<'source>(
+    condition: Expression<'source>,
+    then_branch: Expression<'source>,
+    else_branch: Expression<'source>,
+    ctx: &mut Context<'source>,
+) -> Result<JmlValue<'source>, EvalError> {
+    let cond_l = condition.l;
+    let cond_r = condition.r;
+    let cond = eval_expr(condition, ctx)?;
+    if !cond.is_bool() {
+        let type_error = TypeError {
+            span: (cond_l, cond_r - cond_l).into(),
+            kind: TypeErrorKind::MismatchedTypes {
+                expected: vec![JmlType::Bool],
```

Comment:
So, only boolean values can be used as a condition for `if` statements?

There is no truthiness or falsiness, found in other languages? Like treat `null`, empty lists, empty strings as false, and everything else as true?

Reply:
Currently, only boolean values can be used as conditions in if statements. However, there are plans to introduce truthiness and falsiness for all types, so that values like null, empty lists, and empty strings would be treated as false, and other values as true.